pkg/tests/ossm/smcp_install_test.go

```diff
@@ -15,219 +15,101 @@
-
-		log.Log.Info("Create SMCP v2.1 in namespace ", meshNamespace)
-		util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV21_template, smcp))
-		util.KubeApplyContents(meshNamespace, smmr)
-
-		// patch SMCP identity if it's on a ROSA cluster
-		if env.Getenv("ROSA", "false") == "true" {
-			util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
-		}
-		log.Log.Info("Waiting for mesh installation to complete")
-		util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 300s`, meshNamespace, smcpName)
-
-		log.Log.Info("Verify SMCP status and pods")
-		util.Shell(`oc get -n %s smcp/%s -o wide`, meshNamespace, smcpName)
-		util.Shell(`oc get -n %s pods`, meshNamespace)
-
-		log.Log.Info("Upgrade SMCP to v2.2")
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV22_template, smcp))
-
-		// patch SMCP identity if it's on a ROSA cluster
-		if env.Getenv("ROSA", "false") == "true" {
-			util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
+func TestSMCPInstall(t *testing.T) {
+	NewTest(t).Id("A1").Groups(Smoke, Full, InterOp, ARM).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		t.Cleanup(func() {
+			oc.DeleteNamespace(t, meshNamespace)
+			SetupNamespacesAndControlPlane()
+		})
+		vars := vars{
+			Name:      smcpName,
+			Namespace: meshNamespace,
 		}
-		log.Log.Info("Waiting for mesh installation to complete")
-		time.Sleep(time.Duration(10) * time.Second)
-		util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 360s`, meshNamespace, smcpName)
-
-		log.Log.Info("Verify SMCP status and pods")
-		msg, _ := util.Shell(`oc get -n %s smcp/%s -o wide`, meshNamespace, smcpName)
-		if !strings.Contains(msg, "ComponentsReady") {
-			log.Log.Error("SMCP not Ready")
-			t.Error("SMCP not Ready")
+		versionTemplates := map[string]string{
+			"2.1": smcpV21_template,
+			"2.2": smcpV22_template,
+			"2.3": smcpV23_template,
+			// "2.4": smcpV24_template,
 		}
-		util.Shell(`oc get -n %s pods`, meshNamespace)
-	})
-
-	t.Run("smcp_test_upgrade_2.2_to_2.3", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Create SMCP v2.2 in namespace ", meshNamespace)
-		util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV22_template, smcp))
-		util.KubeApplyContents(meshNamespace, smmr)
-
-		// patch SMCP identity if it's on a ROSA cluster
-		if env.Getenv("ROSA", "false") == "true" {
-			util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
+		smcpVersion := env.GetDefaultSMCPVersion()
+		_, ok := versionTemplates[smcpVersion]
+		if !ok {
+			t.Errorf("Unsupported SMCP version: %s", smcpVersion)
+			return
```

Comment:
You can replace this with `t.Fatalf` and remove the `return`

Reply:
Actually, you can remove this entire block related to `smcpVersion`, since it isn't used anywhere.

Reply:
Done